back/engines/commercial/user_custom_fields/app/serializers/user_custom_fields/xlsx/age_stats_serializer.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module UserCustomFields
+  module Xlsx
+    module AgeStatsSerializer
+      extend self
+
+      # @param [UserCustomField::AgeStats] age_stats
+      def generate(age_stats)
+        xlsx_columns = compute_columns(age_stats)
+        XlsxService.new.xlsx_from_columns(xlsx_columns, sheetname: 'users_by_age')
+      end
+
+      private
+
+      # @param [UserCustomField::AgeStats] age_stats
+      def compute_columns(age_stats)
+        columns = {
+          'age' => age_ranges_column(age_stats.bins),
+          'user_count' => age_stats.binned_counts,
+          'expected_user_count' => age_stats.expected_binned_counts,
+          'total_population' => age_stats.population_counts
+        }.compact
+
+        # Copy (+dup+) the columns because +add_row+ modifies the column arrays in-place.
+        columns.transform_values!(&:dup)
+        add_row(columns, 'age' => 'unknown', 'user_count' => age_stats.unknown_age_count)
+        columns
+      end
+
+      # @param [Hash<String,Array>] columns
+      # @param [Hash<String,Object>] row
+      def add_row(columns, row)
+        columns.each_key do |k|
+          columns[k] << row[k]
+        end
+      end
+
+      def age_ranges_column(bins)
+        bins = bins.dup
+        bins[0] = 0 if bins.first.nil?
```

Comment:
```suggestion
        bins[0] = 0 unless bins.first
```

Reply:
Do you think it improves the readability? I like to make it explicit that I expect `nil`, and not `false`. But maybe that's obvious here.

Reply:
Do you want to make explicit that `bins.first` is `nil`, or do you want to express that there is no first bin? I assumed the latter, because `bins.first` returns `nil` if there is no element in the array. But maybe you wanted to express that the first element in the array holds `nil.`

Reply:
After looking at the code and the tests again, I understand why you did it. I will close this conversation.

Reply:
> But maybe you wanted to express that the first element in the array holds nil.

Yes, exactly!